test/uk/gov/hmrc/onestopshopforexrates/connectors/DesConnectorSpec.scala

```diff
@@ -0,0 +1,152 @@
+    base = baseCurrency,
+    target = targetCurrency,
+    timestamp = LocalDateTime.now,
+    rates = Seq(CoreRate(LocalDate.now, rate))
+  )
+
+  "headers" - {
+
+    "generate the correct Date header format required" in {
+      val dateFormat = DateTimeFormatter
+        .ofPattern("EEE, dd MMM yyyy HH:mm:ss z", Locale.ENGLISH)
+        .withZone(ZoneId.of("GMT"))
+
+      val connector = application.injector.instanceOf[DesConnector]
+
+      connector.headers.filter(item => item._1 == "Date").map { item =>
+        Try(dateFormat.parse(item._2)).isSuccess mustBe true
+      }
+    }
+
+    "generate the correct Correlation ID header required" in {
+      val connector = application.injector.instanceOf[DesConnector]
+      connector.headers must contain("X-Correlation-ID" -> connector.acknowledgementReference.toString)
+    }
+
+    "generate the correct Accept header required" in {
+      val connector = application.injector.instanceOf[DesConnector]
+      connector.headers must contain("Accept" -> "application/json")
+    }
+
+    "generate the correct X-Forwarded-Host header required" in {
+      val connector = application.injector.instanceOf[DesConnector]
+      connector.headers must contain("X-Forwarded-Host" -> "MDTP")
+    }
+
+    "generate the correct Content-Type header required" in {
+      val connector = application.injector.instanceOf[DesConnector]
+      connector.headers must contain("Content-Type" -> "application/json")
+    }
+  }
+
+  "postRates" - {
+
+    "must return OK when rates are successfully sent to core" in {
+
+      val url = "/one-stop-shop-returns-stub/oss/referencedata/v1/exchangerate"
+
+      running(application) {
+        val connector = application.injector.instanceOf[DesConnector]
+
+        val responseBody = Json.toJson(exchangeRateRequest).toString
+
+        server.stubFor(post(urlEqualTo(url)).willReturn(ok().withBody(responseBody)))
+
+        val result = connector.postLast5DaysToCore(exchangeRateRequest).futureValue
+
+        result mustBe Right()
+      }
+    }
+
+    "must return BadRequest when invalid data sent to core" in {
+
+      val url = "/one-stop-shop-returns-stub/oss/referencedata/v1/exchangerate"
+
+      val errorResponseJson = """{}"""
+
+      running(application) {
+        val connector = application.injector.instanceOf[DesConnector]
+
+        server.stubFor(post(urlEqualTo(url)).willReturn(badRequest().withBody(errorResponseJson)))
+
+        val result = connector.postLast5DaysToCore(exchangeRateRequest).futureValue
+
+        val expectedResponse = CoreErrorResponse(result.left.get.timestamp, result.left.get.transactionId, s"UNEXPECTED_400", errorResponseJson)
+
+        result mustBe Left(expectedResponse)
+      }
+    }
+
+    "must return Conflict Found when data is already present in core" in {
+
+      val url = "/one-stop-shop-returns-stub/oss/referencedata/v1/exchangerate"
+
+      val errorResponseJson = """{}"""
```

Comment:
Can we add a test where we have an example error response and parse it correctly?

Reply:
done now, we changed one of the specs to do that